compiler/semtypinst.nim

```diff
@@ -555,6 +555,14 @@ proc replaceTypeVarsTAux(cl: var TReplTypeVars, t: PType): PType =
 
       for i in countup(0, sonsLen(result) - 1):
         if result.sons[i] != nil:
+          if result.sons[i].kind == tyGenericBody:
+            localError(
+              cl.c.config,
+              t.sym.info,
+              "cannot instantiate '" &
+              typeToString(result.sons[i]) &
+              "' inside of type defintion : '" &
+              t.owner.name.s & "'")
```

Comment:
> is t.owner always not nil? (i guess so, not 100% sure)

This is a good question. @Araq or any core devs know the answer to this?
I think it will always be non-null in the recursive / inheritance case, but idk for sure.

Reply:
A type always has an owner.